benchmark/src/lib.rs

```diff
@@ -103,26 +103,32 @@ impl Benchmarker {
         }
     }
 
-    /// Minting given accounts using self's AC client(s).
-    /// Mint TXNs must be 100% successful in order to continue benchmark.
-    /// Therefore mint_accounts() will panic when any mint TXN is not accepted or fails.
-    /// Known issue: Minting opereations from two different Benchmarker instances
-    /// will fail because they are sharing the same faucet account.
-    pub fn mint_accounts(&mut self, mint_key_file_path: &str, accounts: &[AccountData]) {
-        let mut faucet_account = self.load_faucet_account(mint_key_file_path);
-        self.prev_sequence_numbers
-            .insert(faucet_account.address, faucet_account.sequence_number);
+    /// Initialize the sequence numbers for testing accounts.
+    pub fn register_accounts(&mut self, accounts: &[AccountData]) {
         for account in accounts.iter() {
             self.prev_sequence_numbers
                 .insert(account.address, account.sequence_number);
         }
+    }
 
-        let mint_requests = gen_mint_txn_requests(&mut faucet_account, accounts);
+    /// Minting given accounts using self's AC client(s).
+    /// Mint TXNs must be 100% successful in order to continue benchmark.
+    /// Therefore mint_accounts() will panic when any mint TXN is not accepted or fails.
+    /// Known issue: Minting opereations from two different Benchmarker instances
+    /// will fail because they are sharing the same faucet account.
+    pub fn mint_accounts(
+        &mut self,
+        mint_requests: &[SubmitTransactionRequest],
+        faucet_account: &mut AccountData,
+    ) {
         // Disable client staggering for mint operations.
         let stagger_range_ms = self.stagger_range_ms;
         self.stagger_range_ms = 1;
+        // Reason to clone faucet account: Since we require all minting TXNs to succeed,
+        // faucet_account will not have changes during submiting and waiting.
+        let cloned_faucet_account = faucet_account.clone();
         let (num_accepted, num_committed, _, _) =
-            self.submit_and_wait_txn_committed(&mint_requests, &mut [faucet_account]);
+            self.submit_and_wait_txn_committed(mint_requests, &mut [cloned_faucet_account]);
```

Comment:
what's the issue without cloning the faucet account?

Reply:
because `mint_accounts` only borrowed `faucet_account`, so I cannot move it into a vector that .
Do you know a way to pass faucet to `submit_and_wait_txn_committed()`, which takes a slice of accounts, without first put it into vector?